File: android/word/src/main/java/com/dreampany/word/data/model/Load.java
/*
package com.dreampany.word.data.model;

import androidx.annotation.NonNull;
import androidx.room.Ignore;
import android.os.Parcel;

import com.dreampany.frame.data.model.Base;

*/
/**
 * Created by Hawladar Roman on 6/29/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 *//*

public class Load extends Base {

    private int current;
    private int total;

    @Ignore
    public Load() {

    }

    public Load(int current, int total) {
        this.current = current;
        this.total = total;
    }

    @Ignore
    private Load(Parcel in) {
        super(in);
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }

    public static final Creator<Load> CREATOR = new Creator<Load>() {
        @Override
        public Load createFromParcel(Parcel in) {
            return new Load(in);
        }

        @Override
        public Load[] newArray(int size) {
            return new Load[size];
        }
    };

    public void setCurrent(int current) {
        this.current = current;
    }

    public int forward() {
        this.current++;
        return this.current;
    }

    public void setTotal(int total) {
        this.total = total;
    }

    public int getCurrent() {
        return current;
    }

    public int getTotal() {
        return total;
    }
}
*/

File: android/word/src/main/kotlin/com/dreampany/word/data/model/Load.kt
package com.dreampany.word.data.model

import com.dreampany.framework.data.model.Base
import com.dreampany.framework.util.TimeUtil
import com.dreampany.translation.data.model.TextTranslation
import com.google.common.base.Objects
import kotlinx.android.parcel.Parcelize

/**
 * Created by Roman-372 on 7/17/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Parcelize
data class Load(
    override var time: Long,
    override var id: String,
    var current: Int,
    var total: Int
) : Base() {

    constructor(current: Int, total: Int) : this(TimeUtil.currentTime(), "", current, total) {

    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val item = other as TextTranslation
        return Objects.equal(item.id, id)
    }

    override fun hashCode(): Int {
        return Objects.hashCode(id)
    }
}